qos-monitor/src/main/java/eu/arrowhead/core/qos/QoSMonitorApplicationInitListener.java

```diff
@@ -108,7 +108,14 @@ public PingMonitorManager getPingMonitor() {
 	protected List<CoreSystemService> getRequiredCoreSystemServiceUris() {
 		logger.debug("getRequiredCoreSystemServiceUris started...");
 
-		final List<CoreSystemService> result = new ArrayList<>(8);
+		int requiredCoreSystemServiceByExternalMonitoring = 0;
+		if (monitorType.equals(MonitorProviderType.DEFAULTEXTERNAL)) {
+			requiredCoreSystemServiceByExternalMonitoring += 2;
+		}else if (monitorType.equals(MonitorProviderType.ORCHESTRATEDEXTERNAL)) {
+			requiredCoreSystemServiceByExternalMonitoring += 3;
+		}
+
+		final List<CoreSystemService> result = new ArrayList<>(5 + requiredCoreSystemServiceByExternalMonitoring);
```

Comment:
The last time I've calculated wrong, because of Github hides some lines from me.
![kép](https://user-images.githubusercontent.com/2082221/121323155-dfb7f700-c90f-11eb-964a-0429ebf07a5e.png)
I thought the gatekeeper only adds 3 instead of 5. 

So, your first solution (size of 8) was correct. Sorry. I think you should use that istead of this complicated calculation.

Reply:
done